server/models/recipe.py

```diff
@@ -0,0 +1,139 @@
+
+# Base model the other model(s) will subclass
+Base = declarative_base()
+
+class Recipe(Base, BaseModelMixin):
+    """
+    Some notes
+    - chef_profile -- A relationship to chef profile related to this recipe
+    
+    - ingredients -- A list of ingredients to be converted to a string
+        then back to a list when recipe.get_ingredient_list is called
+    
+    - required_items -- A list of required utensils to be converted to a string
+        then back to a list when recipe.get_utensil_list is called
+    
+    - image_links -- A list of image urls to be converted to a string
+        then back to a list when recipe.get_image_link_list is called
+        This will gaurentee positions in case images are ordered
+        in a certain way for a gallery as well as preventing the need
+        for saving the image as Binary in the database directly.
+    
+    - cuisine -- The category this recipe falls in.
+        **A predefined list of cuisines would be very useful here**
+    
+    - price -- The float that holds the price, to be formatted
+        to $19.00 and accessed with recipe.get_formatted_price()
+    """
+    __tablename__ = "recipe"
+
+    id = Column(Integer, primary_key=True)
+
+    # Relationships
+    chef_profile_id = Column(Integer, ForeignKey('profile.id'))
+    chef_profile = relationship('Profile')
+
+    # All fields are required hence nullable=False
+    # General Info -- No length constraints because they
+    # could all end up being fairly long
+    name = Column(String, nullable=False)
+    description = Column(Text, nullable=False)
+    available = Column(Boolean, nullable=False)
+    cuisine = Column(String, nullable=False)
+    price = Column(Float, nullable=False)
+
+    # These store lists as strings in CSV format
+    ingredients = Column(Text, nullable=False)
+    required_items = Column(Text, nullable=False)
+    image_links = Column(Text, nullable=False)
+
+    # TODO Validation for cuisine which 
+    # requires a predefined list of cuisines
+    # Which will also allow a select drop box to be used
+    # instead of an input box
+
+    @validates('price')
+    def validate_price(self, key, price):
+        if not isinstance(price, float):
+            # Try to convert it to a float
+            try:
+                float_price = float(price)
+                # Reassign price to be the float version
+                price = float_price
+            except:
+                raise ValueError(f"Expected a price as an integer or float but got {type(price)}")
+        return price
+
+    @validates('image_links')
+    def validate_image_links(self, key, image_links):
+        """
+        Make sure we were passed a list
+        Convert it to a string and return it
+        """
+        if not isinstance(image_links, list):
+            raise AssertionError(f"Expected image_links to be of type list but got {type(image_links)}")
+        stringified_links = ",".join(link for link in image_links)
```

Comment:
you can actually just do `",".join(image_links)`

Reply:
Good point, thank you!